src/calendar-app/calendar/model/CalendarModel.ts

```diff
@@ -200,19 +212,161 @@ export class CalendarModel {
 		return calendarInfos
 	}
 
+	public async fetchExternalCalendar(url: string): Promise<string> {
+		if (!this.externalCalendarFacade) throw new Error(`externalCalendarFacade is ${typeof this.externalCalendarFacade} at CalendarModel`)
+		const calendarStr = await this.externalCalendarFacade?.fetchExternalCalendar(url)
+		return calendarStr ?? ""
+	}
+
+	public async handleSyncExternalCalendars() {
+		if (isApp() || isDesktop()) {
+			this.syncExternalCalendars()
+			setInterval(() => {
+				this.syncExternalCalendars()
+			}, EXTERNAL_CALENDAR_SYNC_INTERVAL)
+		}
+	}
+
+	public async syncExternalCalendars(groupSettings: GroupSettings[] | null = null, syncInterval: number = EXTERNAL_CALENDAR_SYNC_INTERVAL) {
+		if (!this.externalCalendarFacade || !locator.logins.isFullyLoggedIn()) {
+			return
+		}
+
+		let existingGroupSettings = groupSettings
+		const userController = this.logins.getUserController()
+
+		const groupRootsPromises: Promise<CalendarGroupRoot>[] = []
+		let calendarGroupRootsList: CalendarGroupRoot[] = []
+		for (const membership of userController.getCalendarMemberships()) {
+			groupRootsPromises.push(this.entityClient.load(CalendarGroupRootTypeRef, membership.group))
+		}
+		calendarGroupRootsList = await Promise.all(groupRootsPromises)
+
+		if (!existingGroupSettings) {
+			const { groupSettings: gSettings } = await locator.entityClient.load(UserSettingsGroupRootTypeRef, userController.user.userGroup.group)
+			existingGroupSettings = gSettings
+		}
+
+		for (const { sourceUrl, group } of existingGroupSettings) {
+			if (!sourceUrl) {
+				continue
+			}
+
+			const lastSync = this.deviceConfig.getLastExternalCalendarSync().get(group)
+			const offset = 1000 // Add an offset to account for cpu speed when storing or generating timestamps
```

Comment:
This would be one second, right? Do we need to be that precise for the sync?

Reply:
During some tests the updates were happening every other interval due to less than 400 ms in difference. So I decided to add this offset to avoid this.